libs/core/schedulers/include/hpx/schedulers/queue_holder_thread.hpp

```diff
@@ -404,7 +404,8 @@ namespace hpx { namespace threads { namespace policies {
         void create_thread(thread_init_data& data, thread_id_ref_type* tid,
             std::size_t thread_num, error_code& ec)
         {
-            if (thread_num != thread_num_)
+            if (thread_num != thread_num_ &&
+                data.initial_state == thread_schedule_state::pending)
```

Comment:
Is this related to the changes here? If yes, how?

Reply:
If the thread is not to be run right away (`thread_num != thread_num_`) we should delay scheduling only if the initial state is pending. This is a follow-on change required by adding thread reference counting (and was missing from #5441). Alternatively I could have thrown an exception here, but I was not sure if this wouldn't break existing code relying on the `shared_priority_scheduler`. Possibly not delaying thread creation in this case (if `initial_state != thread_schedule_state::pending`) seemed to be an acceptable compromise.

Reply:
I think I don't see the connection to reference counting. If this needs to change then so does https://github.com/STEllAR-GROUP/hpx/blob/eb83655fdf307d4f172082092adaa0bf7f513e93/libs/core/schedulers/include/hpx/schedulers/shared_priority_queue_scheduler.hpp#L417-L449. What about `pending_boost`?

Reply:
> I think I don't see the connection to reference counting. If this needs to change then so does

When introducing reference counting we realized that any thread created with `run_now == true` have to be created with `pending` as their initial state as otherwise there wouldn't be any way to keep the newly (delay-)created thread alive. With that PR we introduced an exception that would throw if `!run_now && state != pending`. Without the discussed change we would see that exception. As said, I didn't want to risk possibly breaking existing code at your end, even more as I don't know anything about the `shared_priority_queue_scheduler`.

> https://github.com/STEllAR-GROUP/hpx/blob/eb83655fdf307d4f172082092adaa0bf7f513e93/libs/core/schedulers/include/hpx/schedulers/shared_priority_queue_scheduler.hpp#L417-L449

Excellent catch. The same change needs to be applied here.

> . What about `pending_boost`?

Another excellent catch. This needs to be applied retroactively to the reference counting PR as well.